### PR TITLE
Add Ruby/Rack application example with hot reload

### PR DESCRIPTION
**Pull request by dsalahutdinov**
**Description**

Adds sample Ruby/Rack application with hot reload demonstration

### CODE REVIEW

**Comment by dsalahutdinov** [comment]
@googlebot I signed it!

File: integration/examples/ruby/backend/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM ruby:2.7
+
+WORKDIR /app
+COPY . .
```
Comment (dgageot) [inline integration/examples/ruby/backend/Dockerfile]: Can we do something like:
 + Copy Gemfile*
 + Download the dependencies
 + Copy the rest of the sources

Comment (dsalahutdinov) [reply integration/examples/ruby/backend/Dockerfile]: Hey, @dgageot , thanks for review!
It's done.

File: integration/examples/ruby/backend/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+FROM ruby:2.7
+
+WORKDIR /app
+ADD Gemfile* /app/
```
Comment (dgageot) [inline integration/examples/ruby/backend/Dockerfile]: nit: `ADD Gemfile* ./`

File: integration/examples/ruby/backend/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+RUN bundle install
+
+ADD . /app/
```
Comment (dgageot) [inline integration/examples/ruby/backend/Dockerfile]: nit: `ADD . ./`

File: integration/examples/ruby/skaffold.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+apiVersion: skaffold/v2alpha1
+kind: Config
+build:
+  artifacts:
+  - image: ruby-example
+    context: backend
+    sync:
```
Comment (dgageot) [inline integration/examples/ruby/skaffold.yaml]: Have you tried?

```
sync:
  infer:
  - *.rb
```

It would be nice to demonstrate inferred sync

Comment (dsalahutdinov) [reply integration/examples/ruby/skaffold.yaml]: Yep, I've changed to rack unreloader config and skaffold's option to infer. Works well

**Comment by dsalahutdinov** [comment]
It's done!

**Pull request merged**